src/common/transformations/tests/common_optimizations/shared_ops_optimization.cpp

```diff
@@ -433,3 +433,78 @@ TEST_F(SharedTransformationTestsF, SharedShapeOfTestMixed) {
         model_ref = std::make_shared<Model>(NodeVector{concat}, ParameterVector{input});
     }
 }
+
+namespace {
+template <typename T, typename... NodeArgTypes>
+OutputVector createNodesInMemory(const std::vector<size_t>& node_order_in_memory,
+                                 std::shared_ptr<void>& memory,
+                                 const std::string& node_name_prefix,
+                                 NodeArgTypes&&... node_args) {
```

Comment:
NodeArgTypes&&...
it looks like it's always shared_ptr<Node>, so maybe it's better to change argument type and get rid of std::forward